content/client-lib-development-guide/features.textile

```diff
@@ -504,6 +543,16 @@ h4. Capability - *API not agreed yet*
 * @(TC1)@ This type represents a capability for a key or token
 * @(TC2)@ For now a string representation of the JSON will suffice wherever @capability@ is used
 
+h4. ConnectionDetails
+
+* @(CD1)@ Connection details are optionally passed to the client library in the @CONNECTED@ @ProtocolMessage#connectionDetails@ attribute to inform the client about any constraints it should adhere to, and provide additional meta data about the connection. For example, if a request is made to publish a message that exceeds the @maxMessageSize@, the client library can reject the message immediately before publishing it to Ably
+* @(CD2)@ Attributes available in @ConnectionDetails@:
+** @(CD2a)@ @clientId@ contains the client ID assigned to the token, when present
+** @(CD2b)@ @connectionKey@ is the connection secret key string that is used to resume a connection and its state
+** @(CD2c)@ @maxMessageSize@ is the maximum individual message size in bytes
+** @(CD2d)@ @maxFrameSize@ is the maximum size for a single frame of data sent to Ably. This restriction applies to a @ProtocolMessage@ sent over a realtime connection, or the total body size for a REST request
+** @(CD2e)@ @maxInboundRate@ is the maximum allowable number of requests per second from a client or Ably. In the case of a realtime connection, this restriction applies to the number of @ProtocolMessage@ objects sent, whereas in the case of REST, it is the total number of REST requests per second
```

Comment:
This is accurate but now I read this I'm wondering why we don't count the number of messages instead of the number of protocol messages / api requests.